src/compontens/Blog.js

```diff
@@ -1,33 +1,86 @@
-import React from 'react';
+import React, {useState} from 'react';
 import { Link } from 'react-router-dom';
 import {AiOutlineLike, AiOutlineDislike} from 'react-icons/ai';
 import {CgComment} from 'react-icons/cg';
+import {getTime} from '../hooks/getTime';
+
+export default function Blog({blog, reactions, children}) {
+  const time = getTime(blog.blog_date);
+
+  const renderedReactions = reactions.filter((reaction)=>{
+    return reaction.blog_id === blog.id;
+  })
+
+  const like = renderedReactions.filter((reaction)=>{
+    return reaction.reaction_type === 'like';
+  })
+
+  const dislike = renderedReactions.filter((reaction)=>{
+    return reaction.reaction_type === 'dislike';
+  })
+
+  const [upvote, setUpvote] = useState(like.length);
+  const [downvote, setDownvote] = useState(dislike.length);
+
+  const handleReactionClick = async(reaction_type, purpose)=>{
+    try {
+
+      const body = {
+        blog_id: blog.id,
+        reaction_type,
+        purpose
+      } 
+      const response = await fetch(`${process.env.REACT_APP_BASE_URL}/blogs/reaction`, {
+        method: 'POST',
+        headers: {
+          token: localStorage.token,
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(body)
+      });
+
+      const parsedResponse = await response.json();
+      console.log(parsedResponse);
+      if(parsedResponse) {
+        if(reaction_type === 'like'){
```

Comment:
All these strings, example `"like"`, `"dislike"` are constants they are used all over the project. This is ok in JS, but when you get into other languages like java or python these need to be standardized with an Enum or Constant. Here its ok, just telling you to not do it in other languages :)

Reply:
I know, maybe it should be like const LIKE = 'like', or something.

Reply:
In JS yup